engine/runtime/src/main/java/org/enso/interpreter/node/controlflow/caseexpr/CaseNode.java

```diff
@@ -104,12 +110,16 @@ public Object doMatch(
       for (BranchNode branchNode : cases) {
         branchNode.execute(frame, state, object);
       }
-      CompilerDirectives.transferToInterpreter();
-      throw new PanicException(
-          EnsoContext.get(this).getBuiltins().error().makeInexhaustivePatternMatch(object), this);
+      if (fallthroughProfile.profile(isNested)) {
+        return BranchResult.failure(this);
```

Comment:
If it is nested, it returns failure in match.

Reply:
Correct. This is so that the error does not propagate outside of the case expression.